### PR TITLE
use capabilities on the v1 api

### PR DESCRIPTION
**Pull request by mwhudson**
This actually makes the v1 and v2 guided apis a bit closer, which is nice. This area of the code is not very well tested though!

Needs a rebase but going to stick it up for review / comment anyway.

### CODE REVIEW

File: subiquity/common/types.py [hunk 1]
```diff
@@ -315,42 +313,28 @@ class Disk:
...
-    PREFER_ENCRYPTED = 'prefer-encrypted'
-    PREFER_UNENCRYPTED = 'prefer-unencrypted'
-
+class GuidedCapability(enum.Enum):
+    DIRECT = enum.auto()
+    LVM = enum.auto()
+    LVM_LUKS = enum.auto()
+    CORE_BOOT_ENCRYPTED = enum.auto()
+    CORE_BOOT_UNENCRYPTED = enum.auto()
+    # These two are not valid as GuidedChoiceV2.capability:
```
Comment (dbungert) [inline subiquity/common/types.py:322]: Are these `prefer` capabilities staying or going, eventually?

File: subiquity/ui/views/filesystem/guided.py [hunk 2]
```diff
@@ -327,17 +312,27 @@ def local_help(self):
     def done(self, sender):
         results = sender.as_data()
         choice = None
-        if self.storage_encryption is not None:
+        if self.controller.core_boot_capability is not None:
+            if results.get('use_tpm', sender.tpm_choice.default):
+                capability = GuidedCapability.CORE_BOOT_ENCRYPTED
+            else:
+                capability = GuidedCapability.CORE_BOOT_ENCRYPTED
```
Comment (dbungert) [inline subiquity/ui/views/filesystem/guided.py:319]: Double check this one

File: subiquity/server/controllers/filesystem.py [hunk 3]
```diff
@@ -367,6 +370,14 @@ def guided(self, choice: GuidedChoiceV2):
         self.model.guided_configuration = choice
 
         disk = self.model._one(id=choice.target.disk_id)
+
+        if choice.capability.is_core_boot():
+            assert isinstance(choice.target, GuidedStorageTargetReformat)
+            self.use_tpm = (
+                choice.capability == GuidedCapability.CORE_BOOT_ENCRYPTED)
```
Comment (dbungert) [inline subiquity/server/controllers/filesystem.py:377]: would we also want use_tpm == True for PREFER_ENCRYPTED?

File: subiquity/ui/views/filesystem/guided.py (hunk 2 above)
Comment (mwhudson) [reply subiquity/ui/views/filesystem/guided.py:319]: thanks :)

File: subiquity/common/types.py (hunk 1 above)
Comment (mwhudson) [reply subiquity/common/types.py:322]: We talked about this I think? They will probably go eventually and the list of capabilities end up being replaced by something richer that indicates availability and default.

**Review by dbungert — APPROVED** [review]

**Comment by dbungert** [comment]
Friendly reminder to rebase before merge button :)

**Pull request merged**